### PR TITLE
Update ticket billing functionality

### PR DESCRIPTION
**Pull request by o-psi**
This pull request includes updates to mark tickets as billable and keep track of which tickets have been billed. It also adds the ability to link tickets to invoices and vice versa.

### CODE REVIEW

**Comment by wrongecho** [comment]
Test these changes at: https://billabletickets834.pr-review.itflow.org  
 _(automatic message)_

**Comment by o-psi** [comment]
@johnnyq thoughts?

**Comment by johnnyq** [comment]
@o-psi something broke tinymce here in ticket.php take a peek see if you can find what may have broke it.
Also we do not like to use PHP short hand as it doesnt explain the code very well.
but other than that this is very exciting,
ill review more once those are fixed.

**Comment by o-psi** [comment]
Strange thing, TinyMCE works fine in my dev env, wonder if it is due to the new open_modal javascript? Is that possible with TinyMCE?

I will work on the shorthand, I assume you mean the ?? ::

File: ticket_invoice_add_modal.php [hunk 1]
```diff
@@ -13,26 +18,26 @@
                     
                     <ul class="nav nav-pills nav-justified mb-3">
                         <li class="nav-item">
-                            <a class="nav-link active" data-toggle="pill" href="#pills-create-invoice"><i class="fa fa-fw fa-check mr-2"></i>Create New Invoice</a>
+                            <a class="nav-link <?php echo !$addToExistingInvoice ? 'active' : ''; ?>" data-toggle="pill" href="#pills-create-invoice"><i class="fa fa-fw fa-check mr-2"></i>Create New Invoice</a>
```
Comment (o-psi) [inline ticket_invoice_add_modal.php:21]: Is this the PHP Shorthand in question?

Comment (johnnyq) [reply ticket_invoice_add_modal.php:21]: yes thats it

**Comment by johnnyq** [comment]
what is the show modals js code used for?

File: ticket_invoice_add_modal.php (hunk 1 above)
Comment (o-psi) [reply ticket_invoice_add_modal.php]: @johnnyq This has been fixed

**Comment by johnnyq** [comment]
ah gotchya okay that makes sense.
Still reviewing the code but its looking promising much needed feature

**Comment by johnnyq** [comment]
looks good im gonna reel in and we'll fix thing as they arise if any issues are there.

**Pull request merged**